#**Friends Series Analysis**

##Business Question

Can you identify what makes a great Friends episode?

## Data Dictionary

**Year_of_prod**: The year the episode was produced - *Numeric*

**Season**:	The season to which the episode belongs	- *Numeric*

**Episode Number**:	The number of the episode - *Numeric*

**Episode_Title**:	The title of the episode - *Text*

**Duration**:	The duration of the episode in minutes - *Numeric*

**Summary**:	The summary of the episode - *Text*

**Director**:	The director of the episode - *Text*

**Stars**:	The number of stars the episode received out of 10 - *Decimal*

**Votes**:	The amount of votes the episode received - *Numeric*

##Import Data & Explore

In [1]:
# Import necessary libraries
import pandas as pd

In [4]:
# Loading the data set
friend_df = pd.read_csv('/content/drive/MyDrive/Data Science - Content/Personal Projects/friends_episodes_v3.csv',encoding='latin-1')

In [5]:
#Checking the top of our data
friend_df.head(10)

Year_of_prod  Season  Episode Number  ...        Director  Stars Votes
0          1994       1               1  ...   James Burrows    8.3  7440
1          1994       1               2  ...   James Burrows    8.1  4888
2          1994       1               3  ...   James Burrows    8.2  4605
3          1994       1               4  ...   James Burrows    8.1  4468
4          1994       1               5  ...   Pamela Fryman    8.5  4438
5          1994       1               6  ...  Arlene Sanford    8.1  4274
6          1994       1               7  ...   James Burrows    9.0  5469
7          1994       1               8  ...   James Burrows    8.1  4156
8          1994       1               9  ...   James Burrows    8.2  4038
9          1994       1              10  ...    Peter Bonerz    8.1  4064

[10 rows x 9 columns]

In [7]:
# check the shape of our data set
friend_df.shape

(236, 9)

Our data set has 236 observations and 9 columns/variables

In [10]:
# Let's check the data types
friend_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 236 entries, 0 to 235
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Year_of_prod    236 non-null    int64  
 1   Season          236 non-null    int64  
 2   Episode Number  236 non-null    int64  
 3   Episode_Title   236 non-null    object 
 4   Duration        236 non-null    int64  
 5   Summary         236 non-null    object 
 6   Director        236 non-null    object 
 7   Stars           236 non-null    float64
 8   Votes           236 non-null    int64  
dtypes: float64(1), int64(5), object(3)
memory usage: 16.7+ KB


Our dataset's variable data types align with the data dictionary.

In [11]:
# check for missing values
friend_df.isnull().sum()

Year_of_prod      0
Season            0
Episode Number    0
Episode_Title     0
Duration          0
Summary           0
Director          0
Stars             0
Votes             0
dtype: int64

There are no missing values.Let's check for duplicates.

In [13]:
# chcking for duplicates
friend_df.duplicated().sum()

0

There are no duplicated values/observations.

In [14]:
# check fo unique values
for i in friend_df.columns:
 print(friend_df[i].unique())
 print ("\n")

[1994 1995 1996 1997 1998 1999 2000 2001 2002 2003 2004]


[ 1  2  3  4  5  6  7  8  9 10]


[ 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24
 25]


['The One Where Monica Gets a Roommate: The Pilot'
 'The One with the Sonogram at the End' 'The One with the Thumb'
 'The One with George Stephanopoulos'
 'The One with the East German Laundry Detergent' 'The One with the Butt'
 'The One with the Blackout' 'The One Where Nana Dies Twice'
 'The One Where Underdog Gets Away' 'The One with the Monkey'
 'The One with Mrs. Bing' 'The One with the Dozen Lasagnas'
 'The One with the Boobies' 'The One with the Candy Hearts'
 'The One with the Stoned Guy' 'The One with Two Parts: Part 1'
 'The One with Two Parts: Part 2' 'The One with All the Poker'
 'The One Where the Monkey Gets Away' 'The One with the Evil Orthodontist'
 'The One with the Fake Monica' 'The One with the Ick Factor'
 'The One with the Birth' 'The One Where Rachel Finds Out'
 "The One with Ross's New Girlfri

In [15]:
# Let's lower case the column names and make columns into a single word
friend_df.columns = friend_df.columns.str.lower().str.replace(' ', '_')


In [16]:
# check the columns
friend_df.columns

Index(['year_of_prod', 'season', 'episode_number', 'episode_title', 'duration',
       'summary', 'director', 'stars', 'votes'],
      dtype='object')

## Exploratory Data Analysis

### Univariate Analysis

In [17]:
# Creating a Frequency table of column directorscategories and using the value_counts() function
friend_df.director.value_counts()

Gary Halvorson          54
Kevin Bright            54
Michael Lembeck         24
James Burrows           15
Gail Mancuso            14
Peter Bonerz            12
David Schwimmer         10
Ben Weiss               10
Robby Benson             6
Shelley Jensen           6
Terry Hughes             5
Dana De Vally Piazza     3
Sheldon Epps             3
Pamela Fryman            2
Steve Zuckerman          2
Alan Myerson             2
Roger Christiansen       2
Thomas Schlamme          2
Stephen Prime            1
Sam Simon                1
Todd Holland             1
Paul Lazarus             1
Arlene Sanford           1
Joe Regalbuto            1
Mary Kay Place           1
Ellen Gittelsohn         1
David Steinberg          1
Andrew Tsao              1
Name: director, dtype: int64